### PR TITLE
docstrings for plotting_functions.py

### PR DESCRIPTION
**Pull request by brianpm**
This PR is only for improved documentation of `plotting_functions.py`. 

It modifies all docstrings in the module.

Docstrings follow Numpy conventions: https://numpydoc.readthedocs.io/en/latest/format.html

I think I got everything. A few of them don't have as much info, but I think I pretty fully described the functions that users who will be trying to plug in their own scripts would need to see. 

Literally this just changes the docstrings, so I didn't do any testing because no code is altered.

### CODE REVIEW

**Comment by brianpm** [comment]
Added docstrings for `global_latlon_map`

File: lib/plotting_functions.py [hunk 1]
```diff
@@ -1490,29 +1926,48 @@ def plot_zonal_mean_and_save(wks, case_nickname, base_nickname,
...
+    base_nickname : str
+        short name of `bdata` case, use for annotation
+    case_climo_yrs : list
+        years in the `adata` case, use for annotation
+    baseline_climo_yrs : list:
+        years in the `bdata` case, use for annotation
+    adata : xarray.DataArray
+        data to plot ([lev], [lat], lon).
+        The vertical coordinate (lev) must be pressure levels.
+    bdata : xarray:DataArray
```
Comment (nusbaume) [inline lib/plotting_functions.py:1946]: Typo here:

```suggestion
    bdata : xarray.DataArray
```

**Review by nusbaume — CHANGES_REQUESTED** [review]
This is great work!  Just found some minor typos.

File: lib/plotting_functions.py [hunk 2]
```diff
@@ -1,9 +1,86 @@
...
+mask_land_or_ocean(arr, msk, use_nan=False)
+    Apply a land or ocean mask to provided variable.
+get_central_longitude(*args)
+    Determine central longitude for maps.
+global_average(fld, wgt, verbose=False)
+    pure numpy global average.
+spatial_average(indata, weights=None, spatial_dims=None)
+    Compute spatial average
+wgt_rmse(fld1, fld2, wgt):
+    Calculated the area-weighted RMSE.
```
[the remark this answers is not in the document]
Comment (brianpm) [reply lib/plotting_functions.py:19]: ✅

File: lib/plotting_functions.py [hunk 3]
```diff
@@ -1,9 +1,86 @@
...
+                      mdlfld, obsfld, diffld, **kwargs):
+    Map plots of `mdlfld`, `obsfld`, and their difference, `diffld`.
+pres_from_hybrid(psfc, hya, hyb, p0=100000.):
+    Converts a hybrid level to a pressure
+vert_remap(x_mdl, p_mdl, plev)
+    Interpolates to specified pressure levels.
+lev_to_plev(data, ps, hyam, hybm, P0=100000., new_levels=None, convert_to_mb=False)
+    Interpolate model hybrid levels to specified pressure levels.
+pmid_to_plev(data, pmid, new_levels=None, convert_to_mb=False)
+    Interpolate `data` from hybrid-sigma levels to isobaric levels.
```
[the remark this answers is not in the document]
Comment (brianpm) [reply lib/plotting_functions.py:48]: ✅

File: lib/plotting_functions.py [hunk 4]
```diff
@@ -111,15 +207,26 @@ def mask_land_or_ocean(arr,msk, use_nan=False):
...
-       If any of the arguments is an instance of AdfDiag, then check whether it has a central_longitude in `diag_basic_info`.
-       --> This takes precedence.
-         --> Else: if any of the arguments are scalars in [-180, 360], assumes the FIRST ONE is the central longitude.
-       There are no other possible conditions, so if none of those are met,
-       RETURN the default value of 180.
-
-       This allows a script to, for example, allow a config file to specify, but also have a preference:
-       get_central_longitude(AdfObj, 30.0)
+
+    Can provide multiple arguments.
```
[the remark this answers is not in the document]
Comment (brianpm) [reply lib/plotting_functions.py:211]: ✅

File: lib/plotting_functions.py [hunk 5]
```diff
@@ -184,6 +300,33 @@ def global_average(fld, wgt, verbose=False):
...
+    Notes
+    -----
+    When `weights` is not provided, tries to find sensible values.
+    If there is a 'lat' dimension, use `cos(lat)`. 
+    If there is a 'ncol' dimension, looks for `area` in `indata`.
+    Otherwise, set to equal weights.
+
+    Makes an attempt to identify the spatial variables when `spatial_dims` is None.
+    Will average over `ncol` if present, and then will check for `lat` and `lon`.
+    When none of those three are found, raise and AdfError.
```
[the remark this answers is not in the document]
Comment (brianpm) [reply lib/plotting_functions.py:328]: ✅

File: lib/plotting_functions.py [hunk 6]
```diff
@@ -341,6 +529,35 @@ def seasonal_mean(data, season=None, is_climo=None):
...
+    Notes
+    -----
+    Currently assumes 'lat' is a dimension and uses `cos(lat)` as weight.
+    Should use `spatial_average`
+
+    See Also
+    --------
+    spatial_average
+
+    """
```
[the remark this answers is not in the document]
Comment (brianpm) [reply lib/plotting_functions.py:560]: #262

File: lib/plotting_functions.py [hunk 7]
```diff
@@ -555,11 +802,47 @@ def plot_map_vect_and_save(wks, case_nickname, base_nickname,
...
+    wks : str or Path
+        output file path
+    case_nickname : str
+        short name for case
+    base_nickname : str
+        short name for base case
+    case_climo_yrs : list
+        list of years in case climatology
+    baseline_climo_yrs : list
+        list of years in base case climatology
```
[the remark this answers is not in the document]
Comment (brianpm) [reply lib/plotting_functions.py:820]: ✅

File: lib/plotting_functions.py [hunk 8]
```diff
@@ -748,9 +1031,32 @@ def plot_map_and_save(wks, case_nickname, base_nickname,
...
```
[the remark this answers is not in the document]
Comment (brianpm) [reply lib/plotting_functions.py:1045]: ✅

File: lib/plotting_functions.py [hunk 9]
```diff
@@ -748,9 +1031,32 @@ def plot_map_and_save(wks, case_nickname, base_nickname,
...
+    diffld : xarray.DataArray
+        input difference data
+    kwargs : dict, optional
+        variable-specific options, See Notes
+    
+    Notes
+    -----
+    kwargs expected to be a variable-specific section,
+    possibly provided by an ADF Variable Defaults YAML file.
+    Currently it is inspecte for:
```
[the remark this answers is not in the document]
Comment (brianpm) [reply lib/plotting_functions.py:1059]: ✅

File: lib/plotting_functions.py [hunk 10]
```diff
@@ -934,24 +1240,48 @@ def plot_map_and_save(wks, case_nickname, base_nickname,
...
 def pres_from_hybrid(psfc, hya, hyb, p0=100000.):
-    """
-    Converts a hybrid level to a pressure
-    level using the forumla:
-
-    p = a(k)*p0 + b(k)*ps
-
+    """Calculates pressure field
+
+    pressure derived with the forumla:
```
[the remark this answers is not in the document]
Comment (brianpm) [reply lib/plotting_functions.py:1245]: ✅

File: lib/plotting_functions.py [hunk 11]
```diff
@@ -1207,9 +1591,28 @@ def _meridional_plot_preslon(ax, lon, lev, data, **kwargs):
...
+        axes object to use
+    color : str or mpl color specification
+        color for the curve
+    kwargs : dict, optional
+        plotting options
+
+    Notes
+    -----
+    Checks if there is a `lev` dimesion to determine if
+    it is a lat-pres plot or  a line plot.
```
[the remark this answers is not in the document]
Comment (brianpm) [reply lib/plotting_functions.py:1615]: ✅

File: lib/plotting_functions.py [hunk 12]
```diff
@@ -1221,9 +1624,29 @@ def zonal_plot(lat, data, ax=None, color=None, **kwargs):
...
+        axes object to use
+    color : str or mpl color specification
+        color for the curve
+    kwargs : dict, optional
+        plotting options
+
+    Notes
+    -----
+    Checks if there is a `lev` dimesion to determine if
+    it is a lon-pres plot or  a line plot.
```
[the remark this answers is not in the document]
Comment (brianpm) [reply lib/plotting_functions.py:1649]: ✅

File: lib/plotting_functions.py (hunk 1 above)
Comment (brianpm) [reply lib/plotting_functions.py:1946]: ✅

**Review by nusbaume — APPROVED** [review]
Looks great now, thanks!

**Pull request merged**